src/Analysis/Ast/Impl/Specializations/Typing/Types/GenericTypeParameter.cs

```diff
@@ -33,27 +35,68 @@ public GenericTypeParameter(string name, IPythonModule declaringModule, IReadOnl
+            var callLocation = callExpression?.GetLocation(eval.Module);
+
             if (args.Count == 0) {
-                // TODO: report that at least one argument is required.
-                return declaringModule.Interpreter.UnknownType;
+                eval.ReportDiagnostics(
+                    eval.Module.Uri,
+                    new DiagnosticsEntry(Resources.TypeVarMissingFirstArgument,
+                        callLocation?.Span ?? default,
+                        Diagnostics.ErrorCodes.TypeVarArguments,
+                        Severity.Error, DiagnosticSource.Analysis)
+                );
+
+                return false;
             }
 
             var name = (args[0] as IPythonConstant)?.GetString();
             if (string.IsNullOrEmpty(name)) {
-                // TODO: report that type name is not a string.
-                return declaringModule.Interpreter.UnknownType;
+                var firstArgLocation = callExpression?.Args[0]?.GetLocation(eval.Module);
+                eval.ReportDiagnostics(
+                    eval.Module.Uri,
+                    new DiagnosticsEntry(Resources.TypeVarFirstArgumentNotString,
+                        firstArgLocation?.Span ?? default,
+                        Diagnostics.ErrorCodes.TypeVarArguments,
+                        Severity.Warning, DiagnosticSource.Analysis)
+                );
+
+                return false;
+            }
+
+            // Python gives runtime error when TypeVar has two args
+            // e.g. T = TypeVar('T', int)
+            if (args.Count == 2) {
+                eval.ReportDiagnostics(
+                    eval.Module.Uri,
+                    new DiagnosticsEntry(Resources.TypeVarSingleConstraint,
+                        callLocation?.Span ?? default,
+                        Diagnostics.ErrorCodes.TypeVarArguments,
+                        Severity.Error, DiagnosticSource.Analysis)
+                );
+                return false;
             }
 
+            return true;
+        }
+
+        public static IPythonType FromTypeVar(IArgumentSet argSet, IPythonModule declaringModule, IndexSpan location = default) {
+            if (!TypeVarArgumentsValid(argSet))
+                return declaringModule.Interpreter.UnknownType;
+
+            var args = argSet.Values<IMember>();
+            var name = (args[0] as IPythonConstant)?.GetString();
             var constraints = args.Skip(1).Select(a => {
                 // Type constraints may be specified as type name strings.
                 var typeString = (a as IPythonConstant)?.GetString();
                 return !string.IsNullOrEmpty(typeString) ? argSet.Eval.GetTypeFromString(typeString) : a.GetPythonType();
             }).ToArray();
+
             if (constraints.Any(c => c.IsUnknown())) {
-                // TODO: report that some constraints could not be resolved.
+                // TODO: report that some constraints could not be resolved. 
```

Comment:
Nit: Stray space changed at the end here.

Reply:
Done!